tools/parse_baselines/providers/iperf3.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Implement the DataParser for iperf3 throughput tests."""
+
+import json
+import statistics
+import math
+from collections import defaultdict, Iterator
+from typing import List
+
+from providers.types import DataParser
+
+
+def nesteddict():
+    """Create an infinitely nested dictionary."""
+    return defaultdict(nesteddict)
+
+
+# pylint: disable=R0903
+class Iperf3DataParser(DataParser):
+    """Parse the data provided by the iperf3 throughput tests."""
+
+    # pylint: disable=W0102
+    def __init__(self, data_provider: Iterator):
+        """Initialize the data parser."""
+        self._data_provider = iter(data_provider)
+        self._baselines_defs = [
+            "throughput/total",
+            "cpu_utilization_vcpus_total/value",
+            "cpu_utilization_vmm/value",
+        ]
+        # This object will hold the parsed data.
+        self._data = nesteddict()
+
+    # pylint: disable=R0201
+    def _calculate_baseline(self, data: List[float]) -> dict:
+        """Return the target and delta values, given a list of data points."""
+        avg = statistics.mean(data)
+        stddev = statistics.stdev(data)
+        return {
+            'target': math.ceil(round(avg, 2)),
+            # We add a 3% extra margin, to account for small variations that
```

Comment:
Can we just output the raw numbers and leave the extra margin in the responsibility of the comparison tool?

Seems like we're losing some accuracy and data here by truncating the raw values.

Reply:
This is a good point. Should we make it optional, as in provide a flag for adding extra margins? I think it makes sense to add these extra margins only for `delta_percentage`s.  An extra cmd line argument for parse_baseline tool like `--base-delta-percentage 3` should do the job. @gc-plp , @alindima , I am curious if you think this would be the right thing to do.

Reply:
That may work too. It's just the fact that we're hardcoding things that's going to chase us later.

Reply:
The course of action in my opinion is to simply remove the extra percentage from here. This might be needed or not, depending on how stable the test is. Being optional, we can omit it right now and add it once it is needed, through a specific support that lets user increase the delta percentages by an extra margin. @gc-plp , @alindima , let me know if you agree with this course of action. This is the only thing left on my plate before approving the PR.

Reply:
The extra percentage is rather random and should be optional. The 3% value seems chosen at random and may work for some tests while it may fail for some others as @iulianbarbu said.

Reply:
This X% extra value is totally optional. Every test parser will implement the `DataParser` abstract class. It is only added for the tcp vsock throughput tests

Reply:
moved this to a constant